src/sentry/deletions/defaults/group.py

```diff
@@ -94,32 +115,26 @@ def chunk(self) -> bool:
                 {"organization_id": self.groups[0].project.organization_id} if self.groups else None
             ),
         )
-        if not events:
-            # Remove all group events now that their node data has been removed.
-            for project_id, group_ids in project_groups.items():
-                eventstream_state = eventstream.backend.start_delete_groups(project_id, group_ids)
-                eventstream.backend.end_delete_groups(eventstream_state)
-            return False
```

Comment:
This block is moved higher up in the code.

<img width="878" alt="image" src="https://github.com/user-attachments/assets/49cdfb7c-1a26-489b-8c17-ef0a8ede5ae8">